models/spring-ai-ollama/src/main/java/org/springframework/ai/ollama/api/OllamaApi.java

```diff
@@ -374,19 +374,44 @@ public enum Role {
 			/**
 			 * Assistant message type. Usually the response from the model.
 			 */
-			@JsonProperty("assistant") ASSISTANT;
+			@JsonProperty("assistant") ASSISTANT,
+			/**
+			 * Tool message type.
+			 */
+			@JsonProperty("tool") TOOL;
+		}
 
+		/**
+		 * The relevant tool call.
+		 *
+		 * @param function The function definition.
+		 */
+		@JsonInclude(Include.NON_NULL)
+		public record ToolCall(
+				@JsonProperty("function") ToolCallFunction function) {
+		}
+
+		/**
+		 * The function definition for a tool call.
+		 *
+		 * @param name The name of the function.
+		 * @param arguments The arguments that the model expects you to pass to the function.
+		 */
+		@JsonInclude(Include.NON_NULL)
+		public record ToolCallFunction(
+				@JsonProperty("name") String name,
+				@JsonProperty("arguments") Map<String,Object> arguments) {
```

Comment:
This is very similar to the OpenAI API. The main difference is that OpenAI returns arguments as a JSON String, whereas Ollama returns them in a Map.

Reply:
The other difference is that Ollama doesn't currently handle IDs for ToolCalls. Perhaps they will add that in the future.